synapse/config/oidc_config.py

```diff
@@ -190,6 +191,19 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
         #           When rendering, the Jinja2 templates are given a 'user' variable,
         #           which is set to the claims returned by the UserInfo Endpoint and/or
         #           in the ID Token.
+        #   It is possible to configure Synapse to only allow logins if certain attributes
+        #   match particular values in the OIDC userinfo. The requirements can be listed under
+        #   `attribute_requirements` as shown below. All of the listed attributes must
+        #   match for the login to be permitted. Note that fields listed may need to be added
+        #   to the userinfo return in the OIDC provider, by extending or adding a scope.
+        #   If the attributes listed are not part of the default scopes, these scopes may
+        #   also need to be added to the `scopes` section of the OIDC config.
```

Comment:
I think this somewhat repeats the same thing twice, maybe this is a bit clearer?

```suggestion
        #   `attribute_requirements` as shown below. All of the listed attributes must
        #   match for the login to be permitted. Additional attributes can be added to
        #   userinfo by expanding the `scopes` section of the OIDC config to retrieve
        #   additional information from the OIDC provider.
```